### PR TITLE
create service

### PR DESCRIPTION
**Pull request by xlqian**
Create a service that tells you where to park your car!!!!

- [x] Readme.md

- [x] one command build

- [x] test

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Gopkg.toml example
+#
+# Refer to https://golang.github.io/dep/docs/Gopkg.toml.html
```
Comment (kinnou02) [inline Gopkg.toml]: I prefer go module to dep: https://github.com/golang/go/wiki/Modules

File: datamanager.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import (
+	"log"
+
+	"github.com/CanalTP/augeas/model"
+	"github.com/hongshibao/go-kdtree"
+)
+
+type DataManager struct {
+	carParksMap map[string]*model.CarPark
```
Comment (kinnou02) [inline datamanager.go]: It's better to not use pointer if you don't need them

File: datamanager.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	log.Print("Finish building data")
+
+	return &dm
+}
+
+func (dm *DataManager) GetAllCarParks() []*model.CarPark {
+	return dm.carParksVec
+}
+
+func (dm *DataManager) GetCarParkByID(id *string) []*model.CarPark {
```
Comment (kinnou02) [inline datamanager.go]: Same, the pointer here isn't needed

File: handler.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+func GetCarParkByIDHanlder(dm *DataManager) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		id := c.Param("car_park_id")
+		c.JSON(http.StatusOK, serializer.SerializeCarParks(dm.GetCarParkByID(&id)))
+	}
+}
+
+type durationParams struct {
```
Comment (kinnou02) [inline handler.go]: you can bind the params to a struct: https://github.com/gin-gonic/gin#bind-query-string-or-post-data

File: serializer/serializer.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
+package serializer
+
+import (
+	"math"
+
+	"github.com/CanalTP/augeas/model"
+)
+
+type carParkResposne struct {
```
Comment (kinnou02) [inline serializer/serializer.go]: carParkResponse

File: cmd/augeas/main.go [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+	dm := augeas.NewDataManager(carParks)
+
+	router := gin.Default()
+
+	router.GET("/v0/car_parks", augeas.GetCarParksHanlder(dm))
+	router.GET("/v0/car_parks/:car_park_id", augeas.GetCarParkByIDHanlder(dm))
+	router.GET("/v0/park_duration", augeas.GetParkDurationHandler(dm))
+
+	router.Run(":1337")
```
Comment (nonifier) [inline cmd/augeas/main.go]: Geek !

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+clean: ## Remove temporary files
+	go clean
+
+.PHONY: install
+install: ## install project and it's dependancies, useful for autocompletion feature
+	go install -i
+
+
+.PHONY: version
+version: ## display version of gormungandr
```
Comment (TeXitoi) [inline Makefile]: missed comment

**Pull request merged**